File: app/src/test/java/com/omar/retromp3recorder/app/player/TestAudioPlayer.java
package com.omar.retromp3recorder.app.player;

import javax.inject.Inject;
import javax.inject.Named;

import io.reactivex.Observable;
import io.reactivex.subjects.Subject;

import static com.omar.retromp3recorder.app.di.MockModule.PLAYER_SUBJECT;

public class TestAudioPlayer implements AudioPlayer {

    private final Subject<Event> audioBus;

    @Inject
    TestAudioPlayer(@Named(PLAYER_SUBJECT) Subject<Event> audioBus) {
        this.audioBus = audioBus;
    }

    @Override
    public void playerStop() {

    }

    @Override
    public void playerStart(String voiceURL) {

    }

    @Override
    public Observable<Event> observeEvents() {
        return audioBus;
    }
}

File: app/src/test/java/com/omar/retromp3recorder/app/playback/player/TestAudioPlayer.kt
package com.omar.retromp3recorder.app.playback.player

import com.omar.retromp3recorder.app.di.MockModule
import io.reactivex.Observable
import io.reactivex.subjects.BehaviorSubject
import io.reactivex.subjects.Subject
import javax.inject.Inject
import javax.inject.Named

open class TestAudioPlayer @Inject internal constructor(
    @param:Named(MockModule.PLAYER_SUBJECT) private val audioBus: Subject<AudioPlayer.Event>
) : AudioPlayer {

    private val state = BehaviorSubject.createDefault(AudioPlayer.State.Idle)

    override fun playerStop() {
        state.onNext(AudioPlayer.State.Idle)
    }
    override fun playerStart(voiceURL: String) {
        state.onNext(AudioPlayer.State.Playing)
    }
    override fun observeEvents(): Observable<AudioPlayer.Event> {
        return audioBus
    }

    override val isPlaying: Boolean
        get() = true

    override fun observeState(): Observable<AudioPlayer.State> = state

}
